src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -659,6 +657,42 @@ static void AnimateRootView(CGRect rect)
 			ContainerView.Frame = rect;
 	}
 
+	static void ApplyContentInset(UIScrollView? scrolledView, UIScrollView? lastScrollView)
+	{
+		if (scrolledView is null || lastScrollView is null || ContainerView is null)
+			return;
+
+		var keyboardIntersect = CGRect.Intersect(KeyboardFrame, scrolledView.Frame);
+		nfloat movedContainerDistance = 0;
+
+		// if we are moving or have moved the containerview frame, we need to account for that
+		if (TopViewBeginOrigin != InvalidPoint && ContainerView is not null)
+		{
+			movedContainerDistance = TopViewBeginOrigin.Y - ContainerView.Frame.Y;
+		}
+
+		var frameInWindow = ContainerView!.ConvertRectToView(scrolledView.Frame, null);
```

Comment:
@tj-devel709 I apologize, I deleted my previous comment as it was incorrect.
So if you add a `Padding` to the `ContentPage` you'll realize this doesn't work properly.

To make this work we have to get the position of the frame in the page through the parent, and only then get the position in the window.

```cs
var frameInContainer = ContainerView!.ConvertRectFromView(scrolledView.Frame, scrolledView.Superview);
var frameInWindow = ContainerView!.ConvertRectToView(frameInContainer, null);
```

I've tried this and apparently it works fine in every situation.

Reply:
@tj-devel709 I cannot reproduce anymore and I'm not sure why.
What I know is that at some point I was testing this code, and instead of getting an Y coordinate which included the padding I added to the `ContentPage`, I was getting an Y coordinate which was not considering that padding.
So I'm good for now, anyway I would add some UI test which include a padding at page level, just to be sure that it is always accounted in measurements.